src/main/java/gov/cabinetoffice/gap/adminbackend/services/ApplicationFormSectionService.java

```diff
@@ -118,4 +118,38 @@ public void updateSectionStatus(final Integer applicationId, final String sectio
         this.applicationFormRepository.save(applicationForm);
     }
 
+    public void updateSectionOrder(final Integer applicationId, final String sectionId, final Boolean upOrDown) {
+        AdminSession session = HelperUtils.getAdminSessionForAuthenticatedUser();
+
+        ApplicationFormEntity applicationForm = this.applicationFormRepository.findById(applicationId)
+                .orElseThrow(() -> new NotFoundException("Application with id " + applicationId + " does not exist"));
+
+        if (!session.getGrantAdminId().equals(applicationForm.getCreatedBy())) {
+            throw new AccessDeniedException("User " + session.getGrantAdminId()
+                    + " is unable to access the application form with id " + applicationId);
+        }
```

Comment:
Can use an annotation on a new applicationFormRepository.findById(applicationId) method to remove this along with any other snippets of code in this controller that do the same.

Something _like_:
```
    @PostAuthorize("!returnObject.isEmpty() ? returnObject.get().createdBy.id == authentication.principal.grantAdminId : true")

Reply:
Will refactor this after the feature work is in a good place

Reply:
Looks like its in a good place to me 👀

Reply:
Looks like its in a good place 👀

Reply:
Implemented and fixed all service methods and their underlying tests